### PR TITLE
[nvtt] add patch to preserve /MD compiler switch for release dll build.

### PR DESCRIPTION
**Pull request by LaurensVoerman**
[nvtt] add patch to preserve /MD compiler switch for release dll build.

The cmake/OptimalOptions.cmake in the source package overwrites the CMAKE_(C/CXX)_FLAGS_RELEASE , causing the dll build to use the compiler default (/MT) static runtime linkage. My patch preserves the /MD flag, to build a dll linked with the dynamic runtime library.
Regards, Laurens.

### CODE REVIEW

**Comment by dan-shaw** [comment]
@LaurensVoerman Thanks for the PR! This looks like a bug in `OptimalOptions.cmake`. Can you update this upstream instead?

**Comment by PhoebeHui** [comment]
@LaurensVoerman, do you have a chance to address the review suggestions? Thanks!

**Comment by LaurensVoerman** [comment]
Hi Phoebe,
Sorry, this dropped of my priority list, as I found out OpenSceneGraph (my
main interest) doesn't do anything with nvtt.
I suspect that the problem with the missing /MD switch only applies to the
cmake ninja generator, because I never encountered it before (using the
visual studio generators),
but I did not find the time to test that. As to my "fix", I suspect that
the way the cmake variables are used is simply wrong (in overwriting
x_FLAGS_RELEASE instead of modifying), but I'm not confident enough in
cmake to rewrite it.

Basically, do whatever you like (including sending the patch upstream). I
only wrote it to fix the vcpkg "release build with dynamic linked runtime
lib", but I am confident it doesn't break anything else.
Laurens.

**Comment by JackBoosY** [comment]
Pinging @LaurensVoerman  for response. Is work still being done for this PR?

**Comment by LaurensVoerman** [comment]
No.

**Comment by JackBoosY** [comment]
Closing this PR since it seems that no progress is being made. Please reopen if work is still being done.
